ha-jobs-core/src/main/scala/de/kaufhof/hajobs/LockRepository.scala

```diff
@@ -129,17 +132,27 @@ class LockRepository(session: Session, lockTypes: LockTypes) {
 
   /**
    * Returns a list of Locks (jobType + jobId).
+   * To avoid select * statement without key in where clause,
+   * we do multiple select statements with given primary key
    */
   def getAll()(implicit ec: ExecutionContext): Future[Seq[Lock]] = {
-    val query = select().all().from(Table).setConsistencyLevel(LOCAL_QUORUM)
-    session.executeAsync(query).map(rs =>
-      rs.all().foldLeft(Seq.empty[Lock]) { (res, row) =>
-        val lockTypeName = row.getString(LockTypeCol)
-        Try(lockTypes(lockTypeName))
-          .map(lockType => res :+ Lock(lockType, row.getUUID(LockCol)))
-          .getOrElse(res)
-      }
-    )
+    def query(lockType: LockType) = select().all().from(Table)
+      .where(QueryBuilder.eq(LockTypeCol, lockType.name))
+      .setConsistencyLevel(LOCAL_QUORUM)
+
+    val res = lockTypes.getAll.map { lockType =>
```

Comment:
The same as above, a getLocks(lockType) would make the code a bit more readable.